044-cluster-wide-volume-usage-quota-management.md

```diff
@@ -0,0 +1,342 @@
+due to writes to the upcoming KRaft metadata log.
+
+Currently, when using separate controller-only nodes*, those nodes are not described by `Admin#describeCluster` and we cannot use 
+`Admin#describeLogDirs` against the controllers. So the disk usage of the volume could be invisible.
+
+When using controller+broker mode*, by default the metadata log is kept in the first `log.dirs` directory but could be 
+configured to a custom location, potentially on its own volume. If it was put on a separate volume it would not be
+described in the describeLogDirs response, nor contribute to the volume usage of an existing log dir.
+
+So in some cases with controller+broker mode it would afford some protection, as growth of the metadata log dir could
+cause throttling of topic writes (because they are on the same volume as another log dir).
+
+Even if we had reliable insight into the disk usage (by extending the admin apis for example), of the volume the metadata dir 
+resides on, we cannot take effective measures from the quota plugin. Potentially you could use an Authorization
+plugin instead and block metadata-generating operations.
+
+*tested with kafka 3.3.1
+### High level changes
+
+To better support alternative sources for managing quotas and how they interact with volume usage this proposal
+separates some existing concepts and separates their responsibilities. We envisage the architecture looking like
+this ![Component interactions](images/044-quota-plugin-interactions.png)
+
+#### Quota Source
+
+We propose adding a Quota Source concept to the plugin to provide an abstraction through which where we could provide
+external sources of quotas, like pull them from an external system.
+
+#### Throttle Factor
+
+We introduce the concept of throttle factor. This is a factor in the range (0.0, 1.0) that is applied to a quota to
+calculate the final byte rate limit. 
+
+This is a concept that was already implicit in the current calculations, but we want to name it so that we can use it in 
+metric names and provide an extension point in case we want to externalise the Throttle Factor.
+
+#### Throttle Factor Source
+
+We propose adding a Throttle Factor Source concept to the plugin to provide an abstraction through which we could
+provide alternative sources of throttle factors, such as pull the factor from a single decision-making broker.
+
+#### Volume
+
+The storage quotas operate on observations about a **Volume** with these characteristics:
+1. `logDir`: the path of the logdir
+2. `brokerId`
+3. `totalBytes`
+4. `availableBytes`
+5. `usedBytes` (totalBytes - availableBytes)
+6. `availableRatio` (availableBytes/totalBytes)
+
+#### Volume Source
+
+The quota plugin operates on observations about Volumes, the **Volume Source** determines where
+we obtain those observations. Values are:
+1. `local`: we inspect this brokers local log dirs to discover the state of their Volume
```

Comment:
What is the value of this? Just backwards compatibility? Or something else as well?